operator/controllers/actionsrunner/util/converters.go

```diff
@@ -360,22 +360,41 @@ func ToJob(actionsRunner *inlocov1alpha1.ActionsRunner, actionsRunnerJob *inloco
 	return &job, nil
 }
 
-func filterResourceRequirements(source corev1.ResourceRequirements, resourceNames ...corev1.ResourceName) corev1.ResourceRequirements {
-	resourceRequirements := corev1.ResourceRequirements{
-		Limits:   corev1.ResourceList{},
-		Requests: corev1.ResourceList{},
+func withRuntimeAffinity(affinity *corev1.Affinity) *corev1.Affinity {
+	if affinity == nil {
+		affinity = &corev1.Affinity{}
 	}
 
-	for _, resourceName := range resourceNames {
-		if val, ok := source.Limits[resourceName]; ok {
-			resourceRequirements.Limits[resourceName] = val
-		}
-		if val, ok := source.Requests[resourceName]; ok {
-			resourceRequirements.Requests[resourceName] = val
-		}
+	if affinity.NodeAffinity == nil {
+		affinity.NodeAffinity = &corev1.NodeAffinity{}
 	}
+	nodeAffinity := affinity.NodeAffinity
 
-	return resourceRequirements
+	if nodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil {
+		nodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution = &corev1.NodeSelector{}
+	}
+	nodeSelector := nodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution
+
+	nodeSelector.NodeSelectorTerms = append(nodeSelector.NodeSelectorTerms, corev1.NodeSelectorTerm{
```

Comment:
```
If you specify multiple nodeSelectorTerms associated with nodeAffinity types, then the pod can be scheduled onto a node if one of the nodeSelectorTerms can be satisfied.
```
https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/

Is this the intended behavior?

Reply:
Not actually. I will change it to append/override `kubernetes.io/os` and `kubernetes.io/arch` on existent NodeSelectorTerms and if none exists I will create one with the requirements.